src/napari/_qt/layer_controls/widgets/_vectors/qt_edge_color.py

```diff
@@ -0,0 +1,214 @@
+        self._layer.edge_color = color
+
+    def change_edge_color_property(self, property_name: str):
+        """Change edge_color_property of vectors on the layer model.
+        This property is the property the edge color is mapped to.
+
+        Parameters
+        ----------
+        property_name : str
+            property to map the edge color to
+        """
+        mode = self._layer.edge_color_mode
+        try:
+            self._layer.edge_color = property_name
+            self._layer.edge_color_mode = mode
+        except TypeError:
+            # if the selected property is the wrong type for the current color mode
+            # the color mode will be changed to the appropriate type, so we must update
+            self._on_edge_color_mode_change()
+            raise
+
+    def change_edge_color_mode(self, mode: str):
+        """Change edge color mode of vectors on the layer model.
+
+        Parameters
+        ----------
+        mode : str
+            Edge color for vectors. Must be: 'direct', 'cycle', or 'colormap'
+        """
+        old_mode = self._layer.edge_color_mode
+        with self._layer.events.edge_color_mode.blocker():
+            try:
+                self._layer.edge_color_mode = mode
+                self._update_edge_color_gui(mode)
+
+            except ValueError:
+                # if the color mode was invalid, revert to the old mode (layer and GUI)
+                self._layer.edge_color_mode = old_mode
+                self.color_mode_combobox.setCurrentText(old_mode)
+                raise
+
+    def _on_edge_color_mode_change(self):
+        """Receive layer model edge color mode change event & update dropdown."""
+        if not hasattr(self, 'color_mode_combobox'):
+            # Ignore early events i.e when widgets haven't been created yet.
+            return
+
+        with qt_signals_blocked(self.color_mode_combobox):
+            mode = self._layer._edge.color_mode
+            index = self.color_mode_combobox.findText(
+                mode, Qt.MatchFixedString
+            )
+            self.color_mode_combobox.setCurrentIndex(index)
+
+            self._update_edge_color_gui(mode)
+
+    def _on_edge_color_change(self):
+        """Receive layer model edge color  change event & update dropdown."""
+        if (
+            self._layer._edge.color_mode == ColorMode.DIRECT
+            and len(self._layer.data) > 0
+        ):
+            with qt_signals_blocked(self.edge_color_edit):
+                self.edge_color_edit.setColor(self._layer.edge_color[0])
+        elif self._layer._edge.color_mode in (
+            ColorMode.CYCLE,
+            ColorMode.COLORMAP,
+        ):
+            with qt_signals_blocked(self.color_prop_box):
+                prop = self._layer._edge.color_properties.name
+                index = self.color_prop_box.findText(prop, Qt.MatchFixedString)
+                self.color_prop_box.setCurrentIndex(index)
+
+    def _get_property_values(self):
```

Comment:
we should use `features` instead

Reply:
`features` is a replacement of `properties` that uses dataframes instead of simple dictionaries. Using `property` simply maps back to `features` and converts it to a dictionary, for backward compatibility. The migration was never fiinished and properties never fully deprecated, but every time we see `properties` being used in our code, we should try to swap to the new system!

Reply:
The property is old name. We migrate to using features (with better interface).

Reply:
Created issue https://github.com/napari/napari/issues/8156 to followup on this